### PR TITLE
Document `filter` function in array types

### PR DESCRIPTION
**Pull request by darkdrag00nv2**
Added to Cadence in https://github.com/onflow/cadence/pull/2678.

### CODE REVIEW

**Comment by darkdrag00nv2** [comment]
cc: @SupunS @turbolent @dsainati1

File: docs/cadence/language/values-and-types.mdx [hunk 1]
```diff
@@ -1315,6 +1315,69 @@ are available for both variable-sized and fixed-sized or variable-sized arrays.
     let invalidMapFunctionExample = fixedSizedExample.map(functionAcceptingBool)
     ```
 
+-
+    ```cadence
+    access(all)
+    fun filter(_ f: (T: Bool)): [T]
```
Comment (turbolent) [inline docs/cadence/language/values-and-types.mdx:1321]:
```suggestion
    fun filter(_ f: fun(T): Bool): [T]
```

File: docs/cadence/language/values-and-types.mdx [hunk 2]
```diff
@@ -1315,6 +1315,69 @@ are available for both variable-sized and fixed-sized or variable-sized arrays.
...
+            return x % 2 == 0
+        }
+    let invalidFilterFunctionExample = example.filter(functionAcceptingInt64)
+    ```
+
+    `filter` function is also available for fixed-sized arrays:
+
+    ```cadence
+    access(all)
+    fun filter(_ f: (T: Bool)): [T]
```
Comment (turbolent) [inline docs/cadence/language/values-and-types.mdx:1353]:
```suggestion
    fun filter(_ f: fun(T): Bool): [T]
```

**Review by turbolent — APPROVED** [review]
Nice!

File: docs/cadence/language/values-and-types.mdx (hunk 1 above)
Comment (SupunS) [reply docs/cadence/language/values-and-types.mdx:1321]: while at it, can you please also update the signature for the array `map` function above

**Review by SupunS — APPROVED** [review]

File: docs/cadence/language/values-and-types.mdx (hunk 1 above)
Comment (darkdrag00nv2) [reply docs/cadence/language/values-and-types.mdx]: Done, also updated for `map`.

**Pull request merged**